src/Graphics/GraphicsDevice.cs

```diff
@@ -514,15 +516,16 @@ protected virtual void Dispose(bool disposing)
 					 */
 					lock (resourcesLock)
 					{
-						foreach (GCHandle resource in resources.ToArray())
+						GCHandle[] resourceArray = resources.ToArray();
```

Comment:
A note will be good to have in case someone tries to "optimize" it back to the old way